tracer/events.go

```diff
@@ -178,8 +179,36 @@ func startPollingPerfEventMonitor(ctx context.Context, perfEventMap *ebpf.Map,
 					noDataCount.Add(1)
 					continue
 				}
-				triggerFunc(data.RawSample, data.CPU)
+
+				// Keep track of min KTime seen in this batch processing loop
+				trace := t.loadBpfTrace(data.RawSample, data.CPU)
+				if minKTime == 0 || trace.KTime < minKTime {
+					minKTime = trace.KTime
+				}
+				traceOutChan <- trace
+			}
+			// After we've received and processed all trace events,
+			// call SymbolizationComplete if there is a pending oldKTime
+			// that we haven't yet propagated to the rest of the agent.
+			//
+			// This introduces both an upper bound to SymbolizationComplete
+			// call frequency (dictated by pollTicker) but also skips calls
+			// when none are needed (e.g. no trace events have been read).
+			//
+			// We use oldKTime instead of minKTime (except when the latter is
+			// smaller than the former) to take into account scheduling delays
+			// that could in theory result in observed KTime going back in time.
+			if oldKTime > 0 {
+				kt := oldKTime
+				if minKTime > 0 && minKTime < kt {
+					// If current minKTime is smaller than oldKTime, use it
+					// instead of oldKTime (and set it to 0 to avoid a repeat).
+					kt = minKTime
+					minKTime = 0
+				}
+				t.TraceProcessor().SymbolizationComplete(kt)
```

Comment:
Without a `kt` temp variable, the code is easier to read.
```suggestion
				if oldKTime <= minKTime {
					t.TraceProcessor().SymbolizationComplete(oldKTime)
				} else {
					// If minKTime is smaller than oldKTime, use it
					// and reset it to avoid a repeat.
					t.TraceProcessor().SymbolizationComplete(minKTime)
					minKTime = 0
				}
```

Reply:
Hard to rework this a bit as the suggested logic was incorrect when `minKTime == 0`.